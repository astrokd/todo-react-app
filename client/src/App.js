import React, { useState, useEffect } from "react";
import { Container } from 'react-bootstrap'
import "./styles.css";
import Header from "./components/header";
import Interface from "./components/interface";
import Footer from "./components/footer";
import ToDoList from "./components/todolist";
import Settings from './context/Settings';
import PaginationSetter from './components/pagination'

import AuthRules from './context/AuthRules'
import LoginForm from './components/LoginForm'
import Auth from './components/Auth'

import useFetch from './hooks/useFetch'

export default function App() {
  const [count,todoItems,error,isLoading,addNewToDoItem,deletionHandler,completionHandler] = useFetch()
  return (
    <AuthRules>
      <Settings>
        <Container className="App">
          <LoginForm />
          <hr />
          <Auth>
            <Header value={`todos:${count}`} />
            <span>This is the ToDo app</span>
            <Auth permission="create">
              <Interface addNewToDoItem={addNewToDoItem} />
            </Auth>
            <ToDoList 
              todoItems={todoItems}
              isLoading={isLoading}
              error={error}
              addNewToDoItem={addNewToDoItem}
              deletionHandler={deletionHandler}
              completionHandler={completionHandler}
            />
            <PaginationSetter />
          </Auth>
          <Footer />
        </Container>
      </Settings>
    </AuthRules>
  );
}
