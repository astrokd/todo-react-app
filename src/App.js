import React, { useState, useEffect } from "react";
import "./styles.css";
import Header from "./components/header";
import Interface from "./components/interface";
import count from "./components/interface";
import Footer from "./components/footer";

export default function App() {
  return (
    <div className="App">
      <Header value={"todos:"} />
      <h4 className="subheader">Add your ToDo items here!</h4>
      <Interface />
      <Footer />
    </div>
  );
}
