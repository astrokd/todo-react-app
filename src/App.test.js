import React from 'react';
import { render } from '@testing-library/react';
import {shallow, mount } from 'enzyme';
import fetchMock from 'fetch-mock-jest';
import waitForExpect from 'wait-for-expect';
import renderer from 'react-test-renderer'

import App from './App';
import Interface from './components/interface';

import { configure } from 'enzyme';
import 'jest-enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('<App />', () => {
  it('renders correctly', () => {
    const todo = renderer.create(<App />).toJSON()
    expect(todo).toMatchSnapshot()
  })

  it('is alive at application start', () => {
    const component = shallow(<App />)
    expect(component.find('Interface')).toExist()
    expect(component.find('ToDoList')).toExist()
  })
})
describe('<Interface/> (enzyme test)', () => {
  beforeEach(() => {
    fetchMock.restore() //so test don't interfere with each other
  })
  it('Interface is alive, submit input exists', () => {
    let interfaceInput = shallow(<Interface />)
    expect(interfaceInput.find({ children: 'To Do:' }).exists()).toBeTruthy()
    expect(interfaceInput.find({ name: 'description' }).exists()).toBeTruthy()
    expect(interfaceInput.find({ name: 'difficulty' }).exists()).toBeTruthy()
    expect(interfaceInput.find({ value: 'Add to do item' }).exists()).toBeTruthy()
  })

  xit('the Add to do item button adds ', async () => {
    const testRESTy =
      [
        { name: 'test 1 RESTy get from api', url: 'http://example.com/1' },
        { name: 'test 2 RESTy get from api', url: 'http://example.com/2' }
      ]
    fetchMock.getAny(JSON.stringify(testRESTy))
    const app = mount(<App />)
    const button = app.find({ children: 'RUN' })
    button.simulate('click')
    await waitForExpect(() => {
      expect(app.state('results')).toEqual(JSON.stringify(testRESTy))
    })
  })
})