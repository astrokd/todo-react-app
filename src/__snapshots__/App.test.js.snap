// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<div
  className="App container"
>
  <header
    className="App-header"
  >
    <h1
      id="header-counter"
    >
      To Do List | 
      todos:0
       
    </h1>
  </header>
  <span>
    This is the ToDo app
  </span>
  <main
    className="Interface"
  >
    <title>
      To Do List | 
      0
    </title>
    <div
      className="App-form"
    >
      <form
        className=""
        onSubmit={[Function]}
      >
        <div
          className="input-group d-flex"
        >
          <div
            className="input-group-prepend"
          >
            <span
              className="input-group-text"
            >
              To Do:
            </span>
          </div>
          <input
            aria-label="Description"
            className="form-control"
            id="description"
            name="description"
            onChange={[Function]}
            placeholder="To Do Description"
            type="text"
          />
          <input
            aria-label="Difficulty"
            className="form-control"
            id="difficulty"
            max="10"
            min="0"
            name="difficulty"
            onChange={[Function]}
            placeholder="To Do Difficulty"
            type="number"
          />
          <input
            className="form-control btn btn-primary"
            type="submit"
            value="Add to do item"
          />
        </div>
      </form>
    </div>
  </main>
  <div
    className="App-todolist"
  >
    
    <table
      className="table table-sm table-striped"
    >
      <thead
        className="thead-dark"
      >
        <tr
          className="d-flex"
        >
          <th
            className="text-left flex-sm-grow-1"
          >
            To Do Description
          </th>
          <th
            className="flex-grow-*"
          >
            Assigned
          </th>
          <th>
            Difficulty
          </th>
          <th>
            Completion Status
          </th>
          <th>
            delete
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
    <button
      className="btn btn-secondary btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Show only incomplete
    </button>
  </div>
  <div>
    Number of items to display per page:
    <input
      max={30}
      min={1}
      onChange={[Function]}
      placeholder={10}
      type="number"
    />
  </div>
  <footer
    className="App-footer"
  >
    <h6>
      JavaScript-401d34-Lab31 To Do List App
    </h6>
  </footer>
</div>
`;
